firebase-web/src/test/java/io/spine/web/firebase/given/MemoizedFirebase.java

```diff
@@ -64,26 +84,48 @@ public void subscribeTo(NodePath nodePath, ChildEventListener listener) {
     @Override
     public void create(NodePath nodePath, NodeValue value) {
         sleepUninterruptibly(writeLatency);
-        writes.add(nodePath);
+        writes.put(nodePath, value);
     }
 
     @Override
     public void update(NodePath nodePath, NodeValue value) {
         sleepUninterruptibly(writeLatency);
-        writes.add(nodePath);
+        writes.put(nodePath, value);
     }
 
     @Override
     public void delete(NodePath nodePath) {
         sleepUninterruptibly(writeLatency);
-        writes.add(nodePath);
+        writes.remove(nodePath);
     }
 
+    /**
+     * Returns a copy of all made read operations.
+     */
     public ImmutableList<NodePath> reads() {
         return ImmutableList.copyOf(reads);
     }
 
-    public ImmutableList<NodePath> writes() {
-        return ImmutableList.copyOf(writes);
+    /**
+     * Returns a copy of all made write operation.
+     */
+    public ImmutableMap<NodePath, NodeValue> writes() {
+        return ImmutableMap.copyOf(writes);
+    }
+
+    /**
+     * Returns a {@code NodeValue} written to a specific {@code path}.
+     *
+     * @throws IllegalStateException
+     *         if no value is present for the path.
```

Comment:
It looks unformatted to me.

Also, we don't put periods in descriptions of parameters and exceptions.

Reply:
That's how the `@throws` block is formatted.